lib/eventFactory.js

```diff
@@ -38,20 +39,21 @@ function splitPRJobs(config) {
 }
 
 /**
- * Get commit jobs to start that are enabled
- * @method getJobsFromCommit
+ * Get triggered jobs to start that are enabled
+ * @method getJobsFromTrigger
  * @param  {Object}   config
  * @param  {Array}    config.jobs                           Array of job objects
  * @param  {Object}   config.pipelineConfig
  * @param  {Array}    [config.pipelineConfig.workflow]      Order of jobs to be run
  * @param  {Object}   config.pipelineConfig.workflowGraph   Object with nodes and edges that represent the order of jobs
- * @param  {String}   config.startFrom                      Startfrom (e.g. ~commit, ~pr, etc)
+ * @param  {String}   config.startFrom                      Startfrom (e.g. ~commit, ~pr, ~sd@123:main, etc)
  * @return {Array}                                          Array of commit jobs to start
  */
-function getJobsFromCommit(config) {
+function getJobsFromTrigger(config) {
     const { jobs, pipelineConfig, startFrom } = config;
+    const isTrigger = EXTERNAL_TRIGGER.test(startFrom);
 
-    if (startFrom !== '~commit') {
+    if (startFrom !== '~commit' && !isTrigger) {
```

Comment:
This means `getJobsFromCommits` won't work anymore?

Reply:
Oh I see. This should handle both? 
The function name is a little confusing to me.